### PR TITLE
Support Headers

### LINKED ISSUE #41: Support curl's -H name=value header field
Needs to support multiple `-H name=value` fields.

### PR DESCRIPTION
**Pull request by freeformz**
Fixes #41 

Decided to re-use the `headers` type and add the 2 methods to support
flag.Value

### CODE REVIEW

**Comment by davecheney** [comment]
@freeformz looks like your test failed because of infuriating yaml shell escaping

File: main.go [hunk 1]
```diff
@@ -149,6 +159,9 @@ func visit(url *url.URL) {
 	if err != nil {
 		log.Fatalf("unable to create request: %v", err)
 	}
+	for _, h := range httpHeaders {
+		req.Header.Add(headerKeyValue(h))
```
Comment (tcnksm) [inline main.go]: Hi, I came from https://github.com/davecheney/httpstat/pull/33 . Let's merge both features.  Can you handle `Host` header here ?  `Host` header can not be set via `req.Header.Add`. It needs to set via `req.Host`...

Comment (davecheney) [inline main.go]: I think this is ok, the host comes from the URL, see #36 for the `-x` flag that lets you send the request to a different iP.

**Comment by davecheney** [comment]
@freeformz @tcnksm in an effort to reduce the number of blocked PRs I'm going to resolve the conflicts and merge this change this evening. Please feel free to send followups/cleanups if there are things I've missed.

**Pull request merged**